doc/rationale.md

```diff
@@ -0,0 +1,39 @@
+# Ignition rationale
+
+Ignition is a distro-agnostic Linux provisioning utility. When a machine boots up for the first time Ignition can manipulate it in various ways to set up the machine for operations.
+
+There are some principles Ignition is designed around. This document aims to explain these principles and how they impact Ignition's design.
+
+## Ignition only runs on the first boot
+
+Ignition is a provisioning tool, not a configuration management tool. Ignition definitely encourages immutable infrastructure, wherein if a user wishes to modify a machine then they should destroy the old machine and provision a new one. This keeps the user's machines in a well known state with relatively simple tooling.
+
+To this end Ignition only runs on the first boot of a machine. Being a general provisioning utility however, there's nothing stopping a user from using Ignition to set up some configuration management tool. This may even make a lot of sense depending on the organization and requirements the user is trying to meet.
+
+## Ignition produces the machine asked for, or no machine at all
+
+Ignition does what it needs to make the system match the state described in the Ignition config. If for any reason Ignition cannot deliver the exact machine that the config asked for, Ignition prevents the machine from booting successfully.
+
+For example if the user wanted a file from `https://example.com/foo.conf` to be downloaded and put on the disk and Ignition is unable to resolve the URL then Ignition will prevent the machine from successfully booting.
+
+## Ignition configs are declarative
+
+Ignition configs describe the state of a system. Ignition configs do _not_ list a series of steps that Ignition should take.
+
+To this end, Ignition configs do not allow users to provide arbitrary logic (including scripts for Ignition to run). Users describe which filesystems they want to exist, what files need to be created, what users should exist, and more. If a user wants more general setup ability, they can have Ignition set up a one-shot systemd unit that runs on first boot.
+
+## Ignition configs should not be written by hand
+
+Ignition configs are intended to not be human friendly, and users should not write Ignition configs by hand. Ignition configs are technically human readable, being JSON, but non-human-readable formats would have worked just as well.
+
+Hand writing Ignition configs encourages users to not validate configs before boot (which ensures wasting both user and machine time), JSON files aren't friendly with newlines (which makes writing configs by hand frustrating), and Ignition configs don't understand the particulars of the OS on which they'll be run (which requires users to have more understanding of the OS than they'd otherwise need).
```

Comment:
This doesn't read very well and I don't follow the intent.

Reply:
I wanted to elaborate on why we don't believe Ignition configs should be written by hand. I dropped this paragraph and added this to the following one:

> This conversion process allows for distribution specific configs to be converted into a distribution agnostic Ignition config, and ensures that the Ignition config gets validated before a user attempts to use it.

Is that clearer?